### ISSUE TITLE
Support list `=` method?

### ISSUE THREAD

**Issue by vanhuyz** [nateware/redis-objects#246]
Currently if we want to set a new value for a list, we have to `clear` it then use `<<`. It would be very convenient if we can use `=` to assign directly. It is a normal behavior ruby list object's behavior.

```rb
class Team
  include Redis::Objects
  list :on_base
  def id
    1
  end
end
```

```rb
>> team = Team.new
>> team.on_base = ['player1', 'player2']
NoMethodError: undefined method `on_base=' for #<Team:0x00007fbba9b0ab04>
```

I found this trouble when I could not find a way to create a test data using FactoryBot.

```rb
FactoryBot.define do
  factory :team do
    on_base {  ['player1', 'player2'] }
  end
end
```

```rb
Failure/Error: let(:team) { build(:team) }

NoMethodError:
   undefined method `on_base=' for #<Team:0x00007ffcca0d2134>
   Did you mean?  on_base
```

**Comment by nateware** [comment]
Seems like a great feature! Could you create a patch with tests? CC @tmsrjs

**Comment by artinboghosian** [comment]
@nateware I started working on a PR for this. After playing with it a bit, it concerns me that you could overwrite an entire redis key by using the setter method. If you accidently type object.my_redis_list = [1, 2, 3] in a console you would have over written all the contents of that key. My thoughts were adding an option for lists, sets and hashes (which are the objects I added setters for) called :writer that you have to set to true in order to opt into the behavior. 

`list :players, writer: true`

Wanted to get your thoughts @nateware.

**Comment by nateware** [comment]
I don’t have a strong opinion. But aren’t there other places in the library
that already do that? And isn’t that the behavior of ruby objects overall?
The = syntax would overwrite any hash or array in ruby. I would keep it
simple until somebody hits this in the wild and files a bug. If that
happens, we could add an option later.

**Comment by tmsrjs** [comment]
Closed by #250
